### PR TITLE
add auth and htt_scheme for Authentication and add connectors module

### PR DESCRIPTION
**Pull request by Kimjingon**
**Which issue(s) this PR fixes**:
***Issue 1***
```release-note
need 'auth', 'http_scheme' and 'user' argument for using ldap
``` 
Fixes for Issue 1 #
```python
#feast_trino/trino.py


from feast_trino.trino_source import TrinoSource
from feast_trino.trino_utils import Trino
from trino.auth import Authentication
...
class TrinoOfflineStore(OfflineStore):
    @staticmethod
    def pull_latest_from_table_or_query(
        config: RepoConfig,
        data_source: DataSource,
        join_key_columns: List[str],
        feature_name_columns: List[str],
        event_timestamp_column: str,
        created_timestamp_column: Optional[str],
        start_date: datetime,
        end_date: datetime,
        user: str = "user",
        auth: Authentication = None,
        http_scheme: str = "http",
    ) 
...

        client = _get_trino_client(config=config, user=user, auth=auth, http_scheme=http_scheme)

...

def get_historical_features(
        config: RepoConfig,
        feature_views: List[FeatureView],
        feature_refs: List[str],
        entity_df: Union[pd.DataFrame, str],
        registry: Registry,
        project: str,
        full_feature_names: bool = False,
        user: str = "user",
        auth: Authentication = None,
        http_scheme: str = None,
    ) 
...
        client = _get_trino_client(config=config, user=user, auth=auth, http_scheme=http_scheme)
...

def _get_trino_client(config: RepoConfig, user: str, auth: Authentication, http_scheme: str) -> Trino:
    client = Trino(
        user=user, # add user because access is denied if the default value, "user", is used.
        catalog=config.offline_store.catalog,
        host=config.offline_store.host,
        port=config.offline_store.port,
        auth=auth, # for using ldap
        http_scheme=http_scheme, #default is http but need https 
    )
    return client
```

```python
#feast_trino/trino_utils.py


...
class Trino:
    def __init__(
        self,
        host: Optional[str] = None,
        port: Optional[int] = None,
        user: Optional[str] = None,
        catalog: Optional[str] = None,
        auth: Optional[Any] = None,
        http_scheme: str = "http",
    ):
        self.host = host or os.getenv("TRINO_HOST")
        self.port = port or os.getenv("TRINO_PORT")
        self.user = user or os.getenv("TRINO_USER")
        self.catalog = catalog or os.getenv("TRINO_CATALOG")
        self.auth = auth or os.getenv("TRINO_AUTH")
        self.http_scheme = http_scheme or os.getenv("TRINO_HTTP_SCHEME")
        self._cursor = None

    def _get_cursor(self) -> Cursor:
        if self._cursor is None:
            self._cursor = trino.dbapi.connect(
                host=self.host,
                port=self.port,
                user=self.user,
                catalog=self.catalog,
                auth=self.auth,
                http_scheme=self.http_scheme
            ).cursor()

        return self._cursor
...
```

***Issue 2***
```release-note
no module name feast_trino.connectors
``` 
Fixes for Issue 2 #
```python

#setup.py
...
    license="MIT License",
    packages=["feast_trino"],
    packages=["feast_trino", "feast_trino.connectors"],
    install_requires=INSTALL_REQUIRE,
    extras_require={
        "ci": CI_REQUIRE,
...
```
if feast_trino.connectors module is not stable, I think all codes for the feast_trino.connectors module in trino.py should be removed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
         "Bug Tracker": "https://github.com/Shopify/feast-trino/issues",
     },
     license="MIT License",
-    packages=["feast_trino"],
+    packages=["feast_trino", "feast_trino.connectors"],
```
Comment (olivierlabreche) [inline setup.py:38]: This should not be necessary as it is a subpackage. Could this be handled instead by adding an `__init__.py` file in `feast_trino.connectors`?

Comment (Kimjingon) [reply setup.py:38]: Thanks! It seems much more appropriate to add __init__.py file in feast_trino.connectors!

Comment (olivierlabreche) [reply setup.py:38]: Are you comfortable making this change?

Comment (Kimjingon) [reply setup.py]: There was an issue where the `__init__.py` file was added, but the sub-package was not created, so the  `setup.py ` file was partially modified.

**Comment by navid-k-gon** [comment]
I Signed the CLA!

**Review by MattDelac — APPROVED** [review]
Do you have any screenshot showing that the auth works as expected ?

**Comment by MattDelac** [comment]
The linter still complains in CI. Which version of Python and black do you use locally ?

CI uses
- Python 3.7.12
- black 19.10b0

**Comment by Kimjingon** [comment]
I fixed again with 
- Python 3.7.12
- black 19.10b0

**Pull request merged**